src/main/java/omero/cmd/graphs/DuplicateI.java

```diff
@@ -660,6 +827,171 @@ private void linkToNewDuplicates() throws GraphException {
         }
     }
 
+    /**
+     * Handles duplication of a fileset, including creating filesystem directories for the new template path and
```

Comment:
This method only handles the on-file duplication or also the database rows? When it exits early, will the Fileset still be created?

Reply:
It'll create `OriginalFile` rows for directories down to the template prefix but not below, those are made later; under the hood it all uses `PublicRepositoryI.makeCheckedDirs` which does `repositoryDao.register` so directories are created on the filesystem at the same time that they get noted in the database. In the event of rollback, not only should the transaction forget those rows, but the new directories should also be removed by the loop in `abort()`.